### DISCUSSION TITLE
Is $(internal:all_ip) an array of strings?

### DISCUSSION THREAD

**Discussion by Jibun-no-Kage** [bitfocus/companion#2997] [Q&A] [answered]
Is $(internal:all_ip) an array of strings?  Or something else?  Trying to set a trigger that checks if the Stream Deck is connected to a computer with a specific IP address, in this case 192.168.1.34.

So I set the trigger event 'becoming true', condition to $(internal:all_ip).includes('192.168.1.34'), action enable 'another' trigger.  So if I understand this triggers logic, when the stream deck, i.e. surface, is connected to the computer with ip address of 192.168.1.34, this trigger should resolve to 'true' state, and action fires.  But this does not seem to work.  The 'includes' directive should work if all_ip is array of ip addresses which are strings.

**Comment by Julusian** [comment, marked as answer]
Its a single string.
But also your syntax is wrong, as expressions aren't true javascript.

There is an `includes` method which will work which has an example in the docs `includes("Companion is great!", "great")`

**Comment by Jibun-no-Kage** [comment]
Not JavaScript... Interesting I have done some rather complex expressions thinking it was JavaScript... and they worked!  Guess I got lucky thus far.  For is if I do... includes($internal:all_ip), "192.168.1.34") and 192.168.1.34... exists returns true.... ok, will give it a shot.

**Comment by Jibun-no-Kage** [comment]
@Julusian What is really odd... is that the includes() returns 'true' if the sub-string is found, but empty string when not found.  That was really odd IMHO.  But I worked around the odd return result, so now, I have a trigger that is device/machine specific, i.e. i can enable and disable connections and triggers based on where the stream desk is connected, and have one configuration file across all the devices the stream deck may be connected to at any one time.  This really makes it easy to avoid having unique configuration files for each device, and update each unique device configuration when we make changes common across all devices.

**Comment by Jibun-no-Kage** [comment]
Just an after thought... if you create a variable... that has the device name or such?  And that can be a trigger condition, say 'if variable $(interal:HostName) == 'somehostname', that would be a nice feature add.  thinking you could just drop the /etc/hostname value (Linux, MacOS) into a variable?  On Windows %computername%.

**Reply by Julusian** [reply]
there are variables for this in the betas https://github.com/bitfocus/companion/blob/0bcdd0f3826bf7cbc7f2dc55148920c05497a426/companion/lib/Internal/System.js#L149-L156

**Comment by Jibun-no-Kage** [comment]
That is really cool.  Thanks for the information.
